packages/gatsby/src/query/query-compiler.js

```diff
@@ -102,7 +102,14 @@ class Runner {
     // inside <searched_directory>/node_modules.
     const pathRegex = `/{${filesRegex},!(node_modules)/**/${filesRegex}}`
 
-    const modulesThatUseGatsby = await getGatsbyDependents()
+    let modulesThatUseGatsby = await getGatsbyDependents()
+
+    // Exclude unused themes.
+    modulesThatUseGatsby = modulesThatUseGatsby.filter(module =>
+      /gatsby-theme/.test(module.name)
+        ? this.additional.includes(module.path)
+        : true
```

Comment:
Code style nit only: Since `filter` expects a boolean and both `.test` and `.includes` return booleans, this can be .... "simplified" to:

```
!/gatsby-theme/.test(module.name) || this.additional.includes(module.path)
```

For whatever measure of "simple" :p

Unfortunately I can't tell whether this change is correct so I'll let somebody else review the semantics.

Reply:
(If the number of eleents in `this.additional` is expected to be small it might even be better to swap the order since regular expressions are sloooow. But that's very micro.)

Reply:
@pvdz I guess we need to keep that because we only want to exclude themes that are not used. Thanks for the review 👍

Reply:
@pvdz Done 👍